### ISSUE TITLE
Can I intercept audio stream using jack2?

### ISSUE THREAD

**Issue by dstorca** [jackaudio/jack2#572]
I want to intercept an audio stream coming from a voice program(e.g. skype, discord, etc.) before it comes to me(It is not necessary to intercept the voice of specific program. It's okay to intercept the entire system's audio.), and then i manipulate it through python(I want to make the voice clear through deep learning.), and let it come through to my speaker. Is this possible to intercept audio stream using jack2?

In a nutshell,
1. Is there a way to intercept the audio using jack2? If not, is there anything I can refer to?
2. Can I output the manipulated audio using jack2?

**Comment by synthnassizer** [comment]
skype is not jack aware, but you can still do what you want by intercepting the system audio in GNU/Linux.  You can basically redirect the pulse audio (assume ubuntu here) to jack.

see option 3, 4
https://jackaudio.org/faq/pulseaudio_and_jack.html

if you use another distro that does not use pulse audio as its audio backend, the alsa redirection may come in handy:
https://jackaudio.org/faq/routing_alsa.html

once you have jack being the audio backend, you can use qjackctl to connect the computer audio to other jack-aware clients (such as your python voice processor) and then grab the processed output and feed it to the soundcard outputs.

I would be very interested to try this voice processor if the code if open source. Is it? If yes do you have a link ?

Good luck!

**Comment by lucianoiam** [comment]
Additionally:

1. On Windows and macOS check JackRouter
2. https://pypi.org/project/py-jack/
